org.adempiere.base/src/org/compiere/model/MOrder.java

```diff
@@ -2195,7 +2195,7 @@ && getC_Payment_ID() == 0 && getC_CashLine_ID() == 0)
 		if (MDocType.DOCSUBTYPESO_OnCreditOrder.equals(DocSubTypeSO)		//	(W)illCall(I)nvoice
 			|| MDocType.DOCSUBTYPESO_WarehouseOrder.equals(DocSubTypeSO)	//	(W)illCall(P)ickup	
 			|| MDocType.DOCSUBTYPESO_POSOrder.equals(DocSubTypeSO)			//	(W)alkIn(R)eceipt
-			|| MDocType.DOCSUBTYPESO_PrepayOrder.equals(DocSubTypeSO)) 
+			|| MDocType.DOCSUBTYPESO_PrepayOrder.equals(DocSubTypeSO) && dt.isAutoGenerateInout()) 
```

Comment:
adding a && after several || is wrong

operator precedence will apply && before the ||

same in line 2220

Reply:
need to add () here for the last condition

Reply:
Added () for the last condition, bud does this mean the new checkboxes will work only for Prepay Order?

Reply:
Either way, you have to add () to make it clear whether it is applicable to all subtype or just prepay order.